src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_client_factory.py

```diff
@@ -0,0 +1,59 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def cf_media(cli_ctx, *_):
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.mediav3 import AzureMediaServices
+    return get_mgmt_service_client(cli_ctx, AzureMediaServices)
+
+
+def get_mediaservices_client(cli_ctx, *_):
+    return cf_media(cli_ctx).mediaservices
+
+
+def _auth_client_factory(cli_ctx, scope=None):
+    import re
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.mgmt.authorization import AuthorizationManagementClient
+    subscription_id = None
+    if scope:
+        matched = re.match('/subscriptions/(?P<subscription>[^/]*)/', scope)
+        if matched:
+            subscription_id = matched.groupdict()['subscription']
+    return get_mgmt_service_client(cli_ctx, AuthorizationManagementClient, subscription_id=subscription_id)
+
+
+def _graph_client_factory(cli_ctx, **_):
+    from azure.cli.core._profile import Profile
+    from azure.cli.core.commands.client_factory import configure_common_settings
+    from azure.graphrbac import GraphRbacManagementClient
+    profile = Profile(cli_ctx=cli_ctx)
+    cred, _, tenant_id = profile.get_login_credentials(
+        resource=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+    client = GraphRbacManagementClient(cred, tenant_id,
+                                       base_url=cli_ctx.cloud.endpoints.active_directory_graph_resource_id)
+    configure_common_settings(cli_ctx, client)
+    return client
```

Comment:
This is used by the service principals commands. We need those methods (_auth_client_factory, _graph_client_factory) to retrieve the clients to manage the service principals for media accounts (create applications, grant access permissions, create secrets, etc). 

Moving those methods to the core might be a good option. Is this something that you would like us to do? In that case, which will be the best place to put them? azure.cli.core.commands.client_factory seems a good place.

Reply:
It would be nice to have them in the core for others to use. We really need to simplify the process of getting customer's their AAD application, Service Principals, and Keys for services that are moving their customers to AAD and off of ACS. 

We are really complicating things when moving our customers off of simple Keys and trying to explain why they need an AAD AP, then a Service Principal, and then get a Key from that... very confusing stuff for our simple minded customers who may not be core Azure devs.

Reply:
This is also related to https://github.com/Azure/azure-cli/pull/6044#issuecomment-381154706